exonum/src/node/mod.rs

```diff
@@ -950,8 +951,14 @@ impl Node {
             node_cfg.service_keypair(),
             ApiSender::new(channel.api_requests.0.clone()),
         );
-        let blockchain = BlockchainBuilder::new(blockchain, node_cfg.consensus.clone())
-            .with_rust_runtime(channel.endpoints.0.clone(), services)
+        let (rust_runtime, genesis_config) = create_rust_runtime_and_genesis_config(
```

Comment:
I'd suggest to refactor `Node::new` for clarity. Ideally, it should accept `GenesisConfig` directly or via a field in `NodeConfig` (since the consensus config is already there). Likewise, service factories should be accepted instead of `InstanceCollection`s.

Reply:
Should we even pass service factories here? Wouldn't it be better to pass `RustRuntime` populated with factories along with other runtimes instead?